src/server/handlers/github.js

```diff
@@ -101,12 +101,19 @@ export const getSnapcraftData = async (repositoryUrl, token) => {
   try {
     const snapcraftYaml = await internalGetSnapcraftYaml(owner, name, token);
     const snapcraftData = {};
-    for (const index of Object.keys(snapcraftYaml)) {
+    for (const index of Object.keys(snapcraftYaml.contents)) {
       if (SNAPCRAFT_INFO_WHITELIST.indexOf(index) >= 0) {
-        snapcraftData[index] = snapcraftYaml[index];
+        snapcraftData[index] = snapcraftYaml.contents[index];
       }
     }
 
+    // copy snapcraft.yaml path from repo into snapcraftData
+    //
+    // XXX we are mixing our custom `path` into data from snapcraft.yaml file
+    // currently there is no `path` defined in snapcraft syntax
+    // https://snapcraft.io/docs/build-snaps/syntax
+    // and also we whitelist only `name`, so no collision should occur
+    snapcraftData.path = snapcraftYaml.path;
```

Comment:
This is a bit questionable part.

`snapcraftData` was meant to keep (whitelisted) contents of snapcraft.yaml (only name ATM). But it's the only reasonable place to put path info.
Right now I just put it in `path` field, because it doesn't collide with anything from real snapcraft.yaml. But in theory, in future it may (if snapcraft.yaml adds some path property and we would whitelist it).

So the question is, do we keep it under `path`, as it's convenient and in the end we are controlling exactly what part of snapcraft.yaml goes into `snapcraftData` or so we want to protect ourselves from possible future by "obfuscating" the prop name as something like `snapcraft_yaml_path` or similar?